### PR TITLE
Synchronize porting guide with `precice-v3`

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
I just did a merge of `master` into `precice-v3` and resolved the resulting merge conflict in the porting guide.

After resolving the merge conflict, I created this branch and did `git checkout precice-v3 pages/docs/couple-your-code/couple-your-co…de-porting-v2-3.md`. The reson for this PR is to be able to review my resolution of the merge conflict.

To avoid this in the future: Generally, we should always keep the porting guide on `master` and `precice-v3` synchronized, because there is no reason to not directly publish changes in the porting guide to master. The purpose of `precice-v3` is to develop the documentation that we cannot publish yet, because it would overwrite the v2 documentation that is still relevant for users.

There are two important rules:

1. If you are only changing the porting guide, make sure to do this on `master`. Don't forget to merge `master` into `precice-v3` (e.g. https://github.com/precice/precice.github.io/pull/288 and https://github.com/precice/precice.github.io/pull/287).
2. If there are changes in the porting guide that result from PRs with documentation updates on `precice-v3`, you should make sure to synchronize the porting guide with `master`.

### Relevant changes to review:

@davidscn https://github.com/precice/precice.github.io/pull/290
@carme-hp https://github.com/precice/precice.github.io/pull/287
@IshaanDesai https://github.com/precice/precice.github.io/commit/91efb0a7ff13cb9372ca94a7b6fa31b01bdb1f52
@MakisH https://github.com/precice/precice.github.io/commit/778ce08fa9196c6f7c17d2f6e3212b539a2d6f0e
@fsimonis https://github.com/precice/precice.github.io/commit/c8740c373cde00da017af13e299d51ff249e9d01

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@davidscn @carme-hp @MakisH @IshaanDesai @fsimonis: If you see something here that is wrong: Please directly fix it on the branch.

**Review by IshaanDesai — APPROVED** [review]
Looks good :+1: thanks for resolving the conflicts.

File: pages/docs/couple-your-code/couple-your-code-porting-v2-3.md [hunk 1]
```diff
@@ -26,19 +26,22 @@ Please add breaking changes here when merged to the `develop` branch.
...
-- Replace `isMeshConnectivityRequired` with `requiresMeshConnectivityFor`.
-- Replace `isGradientDataRequired` with `requiresGradientDataFor`. Instead of the input argument dataID, pass the meshName and dataName.
+- Replace `isActionRequired()` with their respective requirement clause: `requiresInitialData()`, `requiresReadingCheckpoint()` or `requiresWritingCheckpoint()`.
+- Remove `precice::constants::*` and corresponding `#include` statements as they are no longer needed.
+- Remove `markActionFullfiled()`. If `requiresInitialData()`, `requiresReadingCheckpoint()`, or `requiresWritingCheckpoint()` are called, then they are promised to be acted on. Therefore, `markActionFullfiled()` is no longer needed.
+- Replace `isMeshConnectivityRequired` with `requiresMeshConnectivityFor`. Instead of the input argument `meshID`, pass the `meshName`.
+- Replace `isGradientDataRequired` with `requiresGradientDataFor`. Instead of the input argument `dataID`, pass the `meshName` and `dataName`.
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-porting-v2-3.md:36]: I like how this part is broken down to simpler, clearer steps. :+1:
(whoever did that: good job! -- that's @IshaanDesai )

**Review by MakisH — APPROVED** [review]
All changes make sense. I checked all commits, everything seems to be there.

**Pull request merged**